assets/scripts/app.js

```diff
@@ -0,0 +1,245 @@
+const LOG_EVENT_PLAYER_STRONG_ATTACK = 'PLAYER_STRONG_ATTACK';
+const LOG_EVENT_MONSTER_ATTACK = 'MONSTER_ATTACK';
+const LOG_EVENT_HEAL = 'HEAL';
+const LOG_EVENT_GAME_OVER = 'GAME_OVER';
+let battleLog = [];
+
+
+function writeToLog (ev, val, playerHealth, monsterHelath){
+    let logEntry = {
+        event: ev,
+        value: val,
+        finalPlayerHealth: playerHealth,
+        finalMonsterHealth: monsterHelath
+    };
+
+    switch (ev){
+        case LOG_EVENT_MONSTER_ATTACK:
+            logEntry.target = 'Player';
+            break;
+        case LOG_EVENT_PLAYER_ATTACK:
+            logEntry.target = 'Player';
+            break;
+        case LOG_EVENT_PLAYER_STRONG_ATTACK:
+            logEntry = {
+                event: ev,
+                value: val,
+                target: 'Monster',
+                finalPlayerHealth: playerHealth,
+                finalMonsterHealth: monsterHelath
+            };
+            break;
+        case LOG_EVENT_HEAL:
+            logEntry = {
+                event: ev,
+                value: val,
+                target: 'Player',
+                finalPlayerHealth: playerHealth,
+                finalMonsterHealth: monsterHelath
+            };
+            break;
+        case LOG_EVENT_GAME_OVER:
+            logEntry = {
+                event: ev,
+                value: val,
+                finalPlayerHealth: playerHealth,
+                finalMonsterHealth: monsterHelath
+            };
+            break;
+        default:
+            logEntry = {};
+    }
+
+
+    // if (ev === LOG_EVENT_MONSTER_ATTACK){
+    //     logEntry.target = 'Player';
+    // } else if (ev === LOG_EVENT_PLAYER_ATTACK){
+    //     logEntry.target = 'Monster'
+    // } else if (ev === LOG_EVENT_PLAYER_STRONG_ATTACK){
+    //     logEntry = {
+    //         event: ev,
+    //         value: val,
+    //         target: 'Monster',
+    //         finalPlayerHealth: playerHealth,
+    //         finalMonsterHealth: monsterHelath
+    //     };
+    // } else if (ev === LOG_EVENT_HEAL){
+    //     logEntry = {
+    //         event: ev,
+    //         value: val,
+    //         target: 'Player',
+    //         finalPlayerHealth: playerHealth,
+    //         finalMonsterHealth: monsterHelath
+    //     };
+    // } else if (ev === LOG_EVENT_GAME_OVER){
+    //     logEntry = {
+    //         event: ev,
+    //         value: val,
+    //         finalPlayerHealth: playerHealth,
+    //         finalMonsterHealth: monsterHelath
+    //     };
+    // }
+    battleLog.push(logEntry);
+  
+}
+
+
+function getMaxLifeValue(){
+    const enteredValue = prompt('Choose a Maximum life', '100');
+    const parsedValue = parseInt(enteredValue);
+
+    if (isNaN(parsedValue) || parsedValue <=0){
+        throw{message: 'Invalid Input, enter a number!'};
+    }
+    return parsedValue;
+}
+
+try {
+    choosenMaxLife = getMaxLifeValue();
+} catch (error) {
+    console.log(error);
```

Comment:
we can use `console.warn` or `console.error` here

Reply:
fixed @faran4engg